src/main/groovy/org/standardout/gradle/plugin/platform/internal/FileBundleArtifact.groovy

```diff
@@ -134,13 +134,13 @@ class FileBundleArtifact implements BundleArtifact {
 	 */
 	FileBundleArtifact(BundleArtifact bundle, File sourceBundleFile) {
 		this.file = sourceBundleFile
-		this.id = sourceBundleFile as String
+		this.id = bundle.id + ':sources'
 		this.source = true
 		this.wrap = false // wrapping is done implicitly
 		
 		bndConfig = null
 		
-		version = modifiedVersion = bundle.version
+		version = modifiedVersion = VersionUtil.toOsgiVersion(bundle.version).toString()
```

Comment:
I think in all the implementations modifiedVersion is final and thus determined in the constructor already.